### ISSUE TITLE
`where` operators missing autograd kernel registration, causing grad_norm=0 during training

### ISSUE THREAD

**Issue by physics31415926** [flagos-ai/FlagGems#1971]
## Description

The `where` operator implementations in FlagGems do not register autograd kernels to the correct `DispatchKey::Autograd` key. This causes PyTorch's autograd to silently fall through during backward pass, resulting in **`grad_norm` being 0** — effectively breaking gradient computation.

## PyTorch Warning

```
UserWarning: aten::where: an autograd kernel was not registered to the Autograd key(s)
but we are trying to backprop through it. This may lead to silently incorrect behavior.
This behavior is deprecated and will be removed in a future version of PyTorch.
If your operator is differentiable, please ensure you have registered an autograd kernel
to the correct Autograd key (e.g. DispatchKey::Autograd, DispatchKey::CompositeImplicitAutograd).
If your operator is not differentiable, or to squash this warning and use the previous behavior,
please register torch::CppFunction::makeFallthrough() to DispatchKey::Autograd.
```

## Environment

- **Tested on:** NVIDIA GPU and Muxi (沐曦) GPU
- **Framework:** [verl](https://github.com/volcengine/verl) (reinforcement learning training)
- **Symptom:** `grad_norm` is 0 during training when FlagGems is enabled

## Workaround

Commenting out the following `where` operator registrations resolves the issue:

```python
# ("where.ScalarOther", where_scalar_other),
# ("where.ScalarSelf", where_scalar_self),
# ("where.self", where_self),
# ("where.self_out", where_self_out),
```

After commenting these out, PyTorch falls back to its native `where` implementation and `grad_norm` returns to normal.

## Expected Behavior

FlagGems's `where` operators should either:
1. Register a proper autograd kernel (if differentiable), or
2. Register `torch::CppFunction::makeFallthrough()` to `DispatchKey::Autograd` to correctly delegate gradient computation back to PyTorch.

## Additional Context

This is a silent correctness bug — training appears to proceed normally but gradients are effectively zeroed out, leading to no learning. The PyTorch warning is the only indicator.
